acme_diags/driver/enso_diags_driver.py

```diff
@@ -0,0 +1,168 @@
+from acme_diags.driver import utils
+
+
+def calculate_nino_index(test, parameter, nino_region=None):
+    if test:
+        data = utils.dataset.Dataset(parameter, test=True)
+    else:
+        data = utils.dataset.Dataset(parameter, ref=True)
+    # type: cdms2.tvariable.TransientVariable
+    sst = data.get_timeseries_variable('SST')
+    if nino_region is None:
+        nino_region = default_regions.regions_specs['NINO34']['domain']
+    sst_nino = sst(nino_region)
+    # Domain average
+    sst_avg = cdutil.averager(sst_nino, axis='xy')
+    # Get anomaly from annual cycle climatology
+    sst_avg_anomaly = cdutil.ANNUALCYCLE.departures(sst_avg)
+    nino_index = sst_avg_anomaly
+    return data, sst, nino_index
+
+
+# TODO: Find a more efficient way to implement this algorithm.
+# See https://www.freecodecamp.org/news/data-science-with-python-8-ways-to-do-linear-regression-and-measure-their-speed-b5577d75f8b/
+def perform_regression(test, parameter, var, region, land_frac, ocean_frac, nino_index, test_sst):
+    # TODO: Do we need to re-make `data` or can we use already defined `data`?
+    if test:
+        data = utils.dataset.Dataset(parameter, test=True)
+    else:
+        data = utils.dataset.Dataset(parameter, ref=True)
+    ts_var = data.get_timeseries_variable(var)
+    domain = utils.general.select_region(region, ts_var, land_frac, ocean_frac, parameter)
+    # Average over selected region, and average
+    # over months to get the yearly mean.
+    cdutil.setTimeBoundsMonthly(domain)
+    # Get anomaly from annual cycle climatology
+    print(domain.shape)
+    anomaly = cdutil.ANNUALCYCLE.departures(domain)
+    nlat = len(anomaly.getLatitude())
+    nlon = len(anomaly.getLongitude())
+    reg_coe = anomaly[0, :, :](squeeze=1)
+    for ilat in range(nlat):
+        print(ilat)
+        for ilon in range(nlon):
+            slope, intercept = genutil.statistics.linearregression(anomaly[:, ilat, ilon],
+                                                                   x=nino_index)
+            reg_coe[ilat, ilon] = slope
+    # Use test.sst_units for both test and ref,
+    # for consistent treatment of degC and C units
+    reg_coe.units = '{}/{}'.format(ts_var.units, test_sst.units)
+    print(reg_coe.shape)
+    # Create netcdf files to save results
+    fout = cdms2.open(var + '_nino3_reg_co.nc', 'w')
+    fout.write(reg_coe)
+    fout.close()
+    return domain, reg_coe
+
+
+def create_single_metrics_dict(values, regrid=False):
+    d = {
+        'min': float(min_cdms(values)),
+        'max': float(max_cdms(values)),
+        'mu': float(mean(values)),
+        'sigma': float(std(values))
+    }
+    mu_2 = d['mu'] ** 2
+    sigma_2 = d['sigma'] ** 2
+    d['rmse'] = math.sqrt(mu_2 + sigma_2)
+    rmse_2 = d['rmse'] ** 2
+    d['fraction'] = mu_2 / rmse_2
+    return d
+
+def create_metrics(ref, test, ref_regrid, test_regrid, diff):
+    """Creates the relevant metrics in a dictionary"""
```

Comment:
It might be good to add confidence level to the regression calculation. Also we may want to re name `mu` and `sigma` to `mean` and `std` just to be consistent with our other diags sets. Let's discuss this later..